File: medium/1525.java
class Solution {
    public int numSplits(String s) {
        Set<Character> leftSet = new HashSet<Character>();
        Set<Character> rightSet = new HashSet<Character>();
        int count = 0;
        int [] left = new int[s.length()];
        int [] right = new int[s.length()];
        for (int i = 0; i < s.length(); i++) {
            char x = s.charAt(i);
            if (!leftSet.contains(x)) {
                leftSet.add(x);
            }
            left[i] = leftSet.size();
        }
        for (int i = s.length()-1; i > 0; i--) {
            char x = s.charAt(i);
            if (!rightSet.contains(x)) {
                rightSet.add(x);
            }
            right[i] = rightSet.size();
            if (right[i] == left[i-1]) {
                count++;
            } else if (right[i] > left[i-1]) {
                return count;
            }
        }
        return count;
    }
}

File: medium/1525.js
// var numSplits = function(s) {
//     let left = new Set();
//     let all = {};
//     let arr = [];
//     let count = 0;
//     let uniq = 0;
//     for (let i = 0; i < s.length; i++) {
//         let char = s[i];
//         if (!all[char]) {
//             all[char] = 0;
//             uniq++;
//         }
//         arr.push(uniq)
//         all[char]++;
//     }
//     let right = [];
//     for (let i = s.length-1; i >= 0; i--) {
//         let char = s[i];
//         left.add(char)
//         right[i] = left.size
//     }
//     right[0] = left.size
//     // console.log(arr)
//     // for (let i = 0; i < right.length; i++) {
//     //     // console.log(arr[i], right[i])
//     //     if (right[i] === arr[i]) {
//     //         count++;
//     //     }
//     // }
//     for (let i = right.length-1; i > 0; i--) {
//         if (right[i] === arr[i-1]) {
//             count++;
//         }
//     }
//     // console.log(right, arr)
//     return count
// };

function numSplits(s) {
    let set = new Set();
    let left = [];
    let count = 0;
    let uniq = 0;
    for (let i = 0; i < s.length; i++) {
        let char = s[i];
        
    }
}




console.log(numSplits("aacaba"))
// should be 2

console.log(numSplits("acbadbaada"))
// [1,2,3,3,4,4,4,4,4,4]
// [,2,1]

// Should be 2